### DISCUSSION TITLE
Changing root boundary. @floating-ui/react

### DISCUSSION THREAD

**Discussion by eusahn** [floating-ui/floating-ui#2372] [Q&A]
I've been looking for easy ways to change the clipping container of the positioned element so that it does not overflow.

What's the easiest way to achieve this?
<img width="108" alt="image" src="https://github.com/floating-ui/floating-ui/assets/19979542/c1cee1e5-db3d-44c8-93c0-a72e7d4f4c67">

You can see in the image the tooltip overflows and therefore gets hidden. Ideal behaviour is to increase the offset from the left.

**Comment by FezVrasta** [comment]
That would be achieved with the prevent overflow middleware.

https://floating-ui.com/docs/detectoverflow#boundary

**Comment by eusahn** [comment]
Here's the custom code I had. Would this be an appropriate use?


````  
const customOverflow = {
    name: 'customOverflow',
    async fn(state) {
      const overflow = await detectOverflow(state, {
        boundary: container,
      })
      return {
        state,
      }
    },
  }
````

And then

````
  const { refs, floatingStyles } = useFloating({
    ....
    middleware: [customOverflow],
  })
````

With this code, I get the values of NaN
<img width="367" alt="image" src="https://github.com/floating-ui/floating-ui/assets/19979542/243f6c3a-9b8b-4eff-9aa8-7aa37a18dbd9">

**Reply by atomiks** [reply]
The best option here is to portal the tooltip element and set a higher z-index on it so it will appear over the top of the element. 

Otherwise, you need to provide the `boundary` to `shift`, not a custom middleware here:

```js
const [boundary, setBoundary] = useState(null);

useFloating({
  middleware: [
    shift({ boundary: boundary || undefined })
  ]
});

<div ref={setBoundary} />
```

**Reply by eusahn** [reply]
Didn't work. Here's what I have.

````
const container = document.querySelector('.container')
const { refs, floatingStyles } = useFloating({
    whileElementsMounted: autoUpdate,
    elements: {
      reference: anchor,
    },
    placement: 'top',
    middleware: [
      offset(10),
      shift({
        boundary: container,
      }),
    ],
})
````

I could just set it to strategy: 'fixed' to implement your former solution, but the latter solution is much better. Any idea why it won't work?

EDIT: Actully it's not even the boundary. Soon as I even pass `shift()` the whole thing doesn't work.
